### PR TITLE
fix: fix prior_batches_processed backfill migration [DET-4006]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This fixes an error in the migration to backfill `steps.prior_batches_processed` that results in incorrect numbers and takes extremely long to run.

## Test Plan
- [x] test query with a few experiments

## Commentary (optional)
I'm going to try to figure out how to make this query faster too, but I'm unsure right now.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by stoksc** [comment]
@neilconway added you, I figure if anyone can make this query faster you could help.

**Comment by davetroiano** [comment]
This lgtm; I'm doubting we need to chase performance.  for the case that was impacted by this, the fix brought the query runtime down to seconds (< 1 minute)

**Comment by davetroiano** [comment]
> Do we need to worry about fixing up an incorrect value that was computed by the old version of this migration?

Good point, we may want a new migration to handle this case, that just runs the same query without `WHERE s.prior_batches_processed IS NULL`

**Comment by stoksc** [comment]
> Good point, we may want a new migration to handle this case, that just runs the same query without WHERE s.prior_batches_processed IS NULL
If we did it as a new migration, people that have already upgraded would be ok but haven't already upgraded would need to go through this bad migration to get to the corrected one, but for some installs can't get through the bad one, so I think it's better to just update this and add a release note with a fixup for those that did install 0.13.0.

**Comment by neilconway** [comment]
> If we did it as a new migration, people that have already upgraded would be ok but haven't already upgraded would need to go through this bad migration to get to the corrected one

I think the idea would be to make two changes:

1. Fix the previous migration in-place, as this PR does right now.
2. Add a new migration which overwrites the value of `prior_batches_processed` unconditionally -- the query Dave proposed would be correct, no? For installs that upgrade from 0.12.x to 0.13.1, this won't make any changes, but this would fix the DBs of folks that did 0.12.x -> 0.13.0 -> 0.13.1

**Comment by stoksc** [comment]
Ah, @neilconway I saw your comment after I pushed, but spoke with @davetroiano about it and did just this. Let me know what you think.

File: master/static/migrations/20200828140013_fixup-steps-prior-batches-processed.up.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+-- This backfills steps.prior_batches_processed, again.
+-- In 0.13.0, a bad migration was released that set this value incorrectly, but that was also
+-- so ineffecient it hung for large installations. If a user skipped 0.13.0, this will
```
Comment (neilconway) [inline master/static/migrations/20200828140013_fixup-steps-prior-batches-processed.up.sql:3]: "inefficient"

File: docs/release-notes/1200-fix-prior-batches-processed-backfilling.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+**Bug Fixes**
+
+- Fix a bug with a database migration in v0.12.13 which caused it to run slow and backfill
```
Comment (neilconway) [inline docs/release-notes/1200-fix-prior-batches-processed-backfilling.txt:5]: The buggy database migration is in `0.13.0`, not `0.12.13`, no?

Comment (stoksc) [reply docs/release-notes/1200-fix-prior-batches-processed-backfilling.txt]: Yep.

**Pull request merged**